Repository/NpmRepository.php

```diff
@@ -147,10 +147,18 @@ protected function createArrayRepositoryConfig(array $packageConfigs)
         $loader = new ArrayLoader();
 
         foreach ($packageConfigs as $version => $config) {
-            $config['version'] = $version;
-            $config = $this->assetType->getPackageConverter()->convert($config);
-            $config = $this->assetRepositoryManager->solveResolutions($config);
-            $packages[] = $loader->load($config);
+            try {
+                $config['version'] = $version;
+                $config = $this->assetType->getPackageConverter()->convert($config);
+                $config = $this->assetRepositoryManager->solveResolutions($config);
+                $packages[] = $loader->load($config);
+            } catch (\UnexpectedValueException $exception) {
+                // Most probably version constraint is broken.
+                // Skip this version and hope that another one will be OK
```

Comment:
There should be a warning or notice, I think.

Reply:
Good point, done